### ISSUE TITLE
File's modification date can't be parsed

### ISSUE THREAD

**Issue by AndyScherzinger** [nextcloud/talk-android#1413]
## Steps to reproduce
1. open a chat and share a file from "Nextcloud"
2. see that all files have the date 1.1.1970 (default/fallback)

### Expected behaviour
Parsing the date

### Actual behaviour
Failing parsing the date

## Device information

**Device:** OP9Pro

**Android version:** 11

**Talk version:** latest master

Findings

see ```ReadFilesystemOperation```calling ```BrowserFile.getModelFromResponse``` which in the background leads to parser use of ```at.bitfire.dav4jvm.property.GetLastModified``` which calls 
```
val date = HttpUtils.parseDate(rawDate)
```
which return null and print the following statements to logcat:
```
com.nextcloud.talk2 W/dav4jvm: Couldn't parse date: Tue, 22 Dec 2020 08:58:05 GMT, ignoring
com.nextcloud.talk2 W/dav4jvm: Couldn't parse Last-Modified date
```
Looking at the code that is called here in the background I would have expected this to work:
```
const val httpDateFormatStr = "EEE, dd MMM yyyy HH:mm:ss 'GMT'"
...
...
DateUtils.parseDate(dateStr, Locale.ROOT,
                httpDateFormatStr,
                "EEE, dd MMM yyyy HH:mm:ss zzz", // RFC 822, updated by RFC 1123 with any TZ
                "EEEE, dd-MMM-yy HH:mm:ss zzz", // RFC 850, obsoleted by RFC 1036 with any TZ.
                "EEE MMM d HH:mm:ss yyyy", // ANSI C's asctime() format
                // Alternative formats.
                "EEE, dd-MMM-yyyy HH:mm:ss z",
                "EEE, dd-MMM-yyyy HH-mm-ss z",
                "EEE, dd MMM yy HH:mm:ss z",
                "EEE dd-MMM-yyyy HH:mm:ss z",
                "EEE dd MMM yyyy HH:mm:ss z",
                "EEE dd-MMM-yyyy HH-mm-ss z",
                "EEE dd-MMM-yy HH:mm:ss z",
                "EEE dd MMM yy HH:mm:ss z",
                "EEE,dd-MMM-yy HH:mm:ss z",
                "EEE,dd-MMM-yyyy HH:mm:ss z",
                "EEE, dd-MM-yyyy HH:mm:ss z",
                /* RI bug 6641315 claims a cookie of this format was once served by www.yahoo.com */
                "EEE MMM d yyyy HH:mm:ss z"
        )
```

Running the String in a manual unit test works just fine, so there must be some runtime issue (jvm, locale, don't know)

**Comment by AndyScherzinger** [comment]
@rfc2822 do you have any clue?
```
<d:response>
<d:href>/remote.php/dav/files/andy/welcome.txt</d:href>
<d:propstat>
<d:prop>
<d:getcontenttype>text/plain</d:getcontenttype>
<d:getcontentlength>164</d:getcontentlength>
<d:getlastmodified>Mon, 10 May 2021 11:47:09 GMT</d:getlastmodified>
<d:getetag>&quot;0d710e9eae1ef46bc72f33b15d8c4a92&quot;</d:getetag>
<d:resourcetype/>
<oc:permissions>RGDNVW</oc:permissions>
<oc:id>00048707ocqbj5czxb7u</oc:id>
<oc:size>164</oc:size>
<oc:favorite>0</oc:favorite>
<oc:owner-id>andy</oc:owner-id>
<oc:owner-display-name>Andy Scherzinger</oc:owner-display-name>
<oc:comments-unread>0</oc:comments-unread>
<nc:mount-type></nc:mount-type>
<nc:has-preview>true</nc:has-preview>
<nc:note></nc:note>
</d:prop>
<d:status>HTTP/1.1 200 OK</d:status>
</d:propstat>
<d:propstat>
<d:prop>
<d:displayname/>
<d:creationdate/>
<nc:is-encrypted/>
</d:prop>
<d:status>HTTP/1.1 404 Not Found</d:status>
</d:propstat>
</d:response>
</d:multistatus>
```
looks good to me and also in the debugger it looks okay. As a sidenote while we use dav4jvm 2.1.2 we bumped commons-lang3 to latest and also use okhttp4.x which should be binary compatible and everything works flawlessly except for the lastmodified date parsing :(

**Comment by AndyScherzinger** [comment]
fixed via #1554
